exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/StorageResources.java

```diff
@@ -204,6 +204,13 @@ public JsonResult createOrUpdatePluginJSON(PluginConfigWrapper plugin) {
   @Consumes(MediaType.APPLICATION_FORM_URLENCODED)
   @Produces(MediaType.APPLICATION_JSON)
   public JsonResult createOrUpdatePlugin(@FormParam("name") String name, @FormParam("config") String storagePluginConfig) {
+    name = name.trim();
+    if (name.isEmpty()) {
+      return message("Error (a storage name cannot be empty)");
+    }
+    if (!name.matches("^[a-zA-Z0-9._-]+$")) {
```

Comment:
Since we don't enforce this on the backend side, I don't think we should enforce this on the frontend side as well.

Reply:
Changed the approach to escape special symbols instead of block them.